cmd/sourced/compose/file/file.go

```diff
@@ -69,6 +70,58 @@ func InitDefault() (string, error) {
 	return activeFilePath, nil
 }
 
+// InitDefaultOverride returns the path of the default docker-compose.override.yml file,
+// if it does not exists, it is created using the same version as the passed base.
+// It returns an error if it docker-compose.override.yml not found, or not created.
+func InitDefaultOverride(base string) (string, error) {
+	composeDirPath, err := dir()
+	if err != nil {
+		return "", err
+	}
+
+	globalOverridePath := filepath.Join(composeDirPath, activeDir, "docker-compose.override.yml")
+
+	_, err = os.Stat(globalOverridePath)
+	if err == nil {
+		return globalOverridePath, nil
+	}
+
+	if !os.IsNotExist(err) {
+		return "", errors.Wrap(err, "could not read the global docker-compose.override.yml file")
+	}
+
+	globalOverridePath, err = createDefaultOverride(base, globalOverridePath)
+	if err != nil {
+		return "", errors.Wrap(err, "could not create the global docker-compose.override.yml file")
+	}
+
+	return globalOverridePath, nil
+
+}
+
+func createDefaultOverride(base, target string) (string, error) {
+	var err error
+	var content []byte
+	var override struct {
+		Version string `yaml:"version"`
```

Comment:
does it fail when versions of main yml and override.yml are different?

Reply:
No, it does not fail if both versions mismatch.
But having hardcoded version in `docker-compose.override.yml` will eventually cause this mismatch, and will force us to update it wherever we update the official `docker-compose.yml`.
Would you prefer it to avoid this _"read official version &rarr; use it in the empty-ish file"_?

Reply:
Eventually means like in a few years? If I understood you correctly minor versions are compatible and it doesn't look like there are any plans for introducing v4.

Reply:
I see you don't care about what's happening in the next few years :p
Let's say that your pov makes sense &rarr; let's simplify.

Reply:
simplified by 9459fe58326117394c5ba69685074f9eb485eecb